homeassistant/components/switchbee/coordinator.py

```diff
@@ -71,4 +71,4 @@ async def _async_update_data(self) -> dict[int, SwitchBeeBaseDevice]:
                 f"Error communicating with API: {exp}"
             ) from SwitchBeeError
 
-        return self.api.devices
+        return self.api.devices  # type: ignore[no-any-return]
```

Comment:
Is the base library not typed? I am surprised that this is necessary.

Reply:
It is typed, but not in a very strict way

Reply:
The library has it as dict. You should change it there to be dict[int, SwitchBeeBaseDevice]

Reply:
I tried to fix the code in the library locally I the dev environment to:

```python
       self._devices_map: dict[int, SwitchBeeBaseDevice] = {}
```

```python
    @property
    def devices(self) -> dict[int, SwitchBeeBaseDevice]:
        return self._devices_map
```

however it still fails on:

```shell
root ➜ /workspaces/core (switchbee_strict_typting ✗) $ mypy homeassistant/components/switchbee/  pylint
homeassistant/components/switchbee/coordinator.py:74: error: Returning Any from function declared to return "Dict[int, Any]"  [no-any-return]
Found 1 error in 1 file (checked 9 source files)
```

Not sure what I missing...

Reply:
I wonder if the package is missing a `py.typed` file

Reply:
It doesn't have

Reply:
As per PEP 561, you should create this empty file in your package, to indicate to users that it is typed.
Currently, mypy only sees it as `Any`

Reply:
For testing, you can use reveal_type like this:
```python
        reveal_type(self.api.devices)
        return self.api.devices
```

In combination with the command line:
`mypy homeassistant/components/switchbee/coordinator.py`

Reply:
Thank you @epenet, I will address it all once the other platforms are merged

Reply:
The package is still not correctly typed.
I am not sure how to solbe, maybe this link helps?
https://stackoverflow.com/questions/70921421/my-python-multi-module-package-wont-distribute-with-a-py-typed-file

Reply:
how do you see its not correctly typed, I see now I don't have the issue return value type in coordinator.py after adding the py.typed file

Reply:
You can read my comment above about reveal_type.
When the package is recognised by mypy the type ignore will be unnecessary

Reply:
But I am not ignoring the type anymore...

Reply:
I still see the line in the PR

Reply:
You are correct, forgot to push it, sorry